### PR TITLE
Initial pull request for lensed SNe

### PR DESCRIPTION
**Pull request by jpierel14**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+numpy
+sncosmo
```
Comment (sibirrer) [inline requirements.txt:2]: can you specify the minimum version number to make sure that the environment is set up well even if people have old version of sncosmo?

File: sim_pipeline/__init__.py [hunk 2]
```diff
@@ -2,3 +2,6 @@
 
 __author__ = """DESC/SLSC"""
 __version__ = '0.1.0'
+
+from .transient_generation import *
```
Comment (sibirrer) [inline sim_pipeline/__init__.py:6]: just wondering whether we need these generic inputs or not.

File: sim_pipeline/sim_pipeline.py [hunk 3]
```diff
@@ -1 +1,23 @@
 """Main module."""
+
+import numpy as np
+import copy
+import matplotlib.pyplot as plt
+
+from .transient_generation import LensedTransientImage,LensedTransientSystem
+
+
+def LensedSystem_from_LensParams(lensed_transients, time_delays, magnifications):
```
Comment (sibirrer) [inline sim_pipeline/sim_pipeline.py:10]: can you document the inputs to this definition? Also, definitions are lower_case_wordings

File: tests/test_sim_pipeline.py [hunk 4]
```diff
@@ -2,23 +2,55 @@
 
 """Tests for `sim_pipeline` package."""
 
-import pytest
+import unittest
```
Comment (sibirrer) [inline tests/test_sim_pipeline.py:5]: can you write the test functions in files and folders that mimic the sim-pipeline file structure such that we all know where exactly to look for a test function for a specific definition

**Review by sibirrer — CHANGES_REQUESTED** [review]
Thanks a lot @jpierel14! I just made some minor comments/request mainly on the structure such that merging and co-development works well. Have not reviewed the code itself. Also let me know if you encounter issues with the testing. Seems that the tests fail for some reason.

File: requirements.txt (hunk 1 above)
Comment (jpierel14) [reply requirements.txt]: Updated

File: sim_pipeline/sim_pipeline.py (hunk 3 above)
Comment (jpierel14) [reply sim_pipeline/sim_pipeline.py:10]: This function is sort of temporary, pending the connection point to the lens sim part so I'll leave it for now

File: tests/test_sim_pipeline.py (hunk 4 above)
Comment (jpierel14) [reply tests/test_sim_pipeline.py]: Updated

**Pull request closed without merging**